cms/server/admin/handlers/contestranking.py

```diff
@@ -78,10 +134,13 @@ def get(self, contest_id, format="online"):
             total_score = 0.0
             partial = False
             for task in self.contest.tasks:
-                t_score, t_partial = task_score(p, task, rounded=True)
-                
-                has_submissions = any(s.task_id == task.id and s.official 
-                                     for s in p.submissions)
+                # Get the cache entry with score and has_submissions
+                cache_entry = get_cached_score_entry(self.sql_session, p, task)
```

Comment:
In case this had to rebuild the cache, is it committed anywhere (releasing the locks)?